src/librustc/ty/layout.rs

```diff
@@ -560,13 +567,27 @@ impl<'a, 'gcx, 'tcx> Struct {
                   repr: &ReprOptions, kind: StructKind,
                   scapegoat: Ty<'gcx>) -> Result<Struct, LayoutError<'gcx>> {
         let packed = repr.packed;
+        let repr_align = repr.align;
```

Comment:
IMO copying fields out of the struct like that is confusing, both should be referred as `repr.foo`.